src/index.js

```diff
@@ -163,9 +172,15 @@ class Dayjs {
     return this
   }
 
-  set(string, int) {
+  set(unitOrDate, int) {
+    if (typeof unitOrDate === 'object' &&
```

Comment:
@Frondor 
> myDayjs.set(new Date(2018, 5, 1)) // you can now pass a Date or Dayjs instance to set!

Could you explain the meaning of you code here, please?  Don't really understand.

Reply:
Just for the record, this is the old PR, the new one is #90 
This code is kind of explained in the test case https://github.com/Frondor/dayjs/blob/5f574850d1cb75c6bd2a8b1684a94346f1d501b7/test/get-set.test.js#L69
In the past, you were limited to set certain time units over an already defined date (`$d`). Now that code checks whether the `unitOrDate` passed is either a time unit, a `Date` object, or a `Dayjs` instance. So you can set the whole date if needed, not just an unit. Useful for inheriting configurations like locale, plugins and default format, and create date objects out of a single source, and keeping the Dayjs immutable since that's how `set` works.

Reply:
set(unitOrDate, int) {
    // if unitOrDate is a Date or Dayjs object
    if (typeof unitOrDate === 'object' &&
      unitOrDate.constructor.name.indexOf(['Date', 'Dayjs'] > -1)) {
      const self = this.clone() // don't mutate this instance
     // set $d date object depending on whether its a native Date object, or a Dayjs instance
      self.$d = unitOrDate.toDate ? unitOrDate.toDate() : unitOrDate
      return self // done
    }
    // if this is not a date object nor an int just fail silently (this code was already here)
    if (!Utils.isNumber(int)) return this
   // unitOrDate is definitely a integer (unit)
    return this.clone().mSet(unitOrDate, int)
  }

Reply:
@Frondor Got it, thanks.